File: backend/src/main/java/com/desarrollo/raffy/dto/GuessProgressDTO.java
package com.desarrollo.raffy.dto;

import java.time.OffsetDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class GuessProgressDTO {

    private int attemptCount;
    private String numbersTried;
    private boolean hasWon;
    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss.SSSX")
    private OffsetDateTime lastAttemptTime;
    private Long durationSeconds;
}

File: frontend/src/app/models/guessprogressDTO.ts
export interface GuessProgressDTO {
    attemptCount: number;
    numbersTried: string;
    hasWon: boolean;
    // lastAttemptTime: Date;
    lastAttemptTime: string;
    durationSeconds: number;
}

export interface DataPlayParticipantDTO {
    eventTitle: string;
    name: string,
    surname: string,
    email: string,
    cellphone: string;
    attemptCount: number;
    numbersTried: string;
    hasWon: boolean;
    lastAttemptTime: string;
    durationSeconds: number;
}